doc/examples/motion_planning_python_api/motion_planning_python_api_tutorial.rst

```diff
@@ -0,0 +1,298 @@
+
+        # plan to goal
+        logger.info("Planning trajectory")
+        plan_result = panda_arm.plan()
+
+        # execute the plan
+        if plan_result:
+                logger.info("Executing plan")
+                panda_arm.execute()
+
+
+Single Pipeline Planning - Robot State
+==================================================
+Next we will plan to a robot state, such a method is quite flexible as we can alter the robot state configuration as we wish (e.g. through setting joint values), here we will just set the robot state to a random configuration for simplicity. We will use the `set_start_state_to_current_state` method to set the start state of the robot to its current state and the `set_goal` method to set the goal state of the robot. We will then plan to the goal state and execute the plan: ::
+
+        # instantiate a RobotState instance using the current robot model
+        robot_model = moveit.get_robot_model()
+        robot_state = RobotState(robot_model)
+
+        # randomize the robot state
+        robot_state.set_to_random_positions()
+
+        # set plan start state to current state
+        panda_arm.set_start_state_to_current_state()
+
+        # set goal state to the initialized robot state
+        logger.info("Set goal state to the initialized robot state")
+        panda_arm.set_goal(robot_state=robot_state)
+
+        # plan to goal
+        logger.info("Planning trajectory")
+        plan_result = panda_arm.plan()
+
+        # execute the plan
+        if plan_result:
+                logger.info("Executing plan")
+                panda_arm.execute()
+
+
+Single Pipeline Planning - Pose Goal
+==================================================
+Another common way to specify a goal state is via a pose goal ros message. Here we demonstrate how to set a pose goal for the end effector of the robot: ::
+
+        # set plan start state to current state
+        panda_arm.set_start_state_to_current_state()
+
+        # set pose goal with PoseStamped message
+        pose_goal = PoseStamped()
+        pose_goal.header.frame_id = "panda_link0"
+        pose_goal.pose.orientation.w = 1.0
+        pose_goal.pose.position.x = 0.28
+        pose_goal.pose.position.y = -0.2
+        pose_goal.pose.position.z = 0.5
+        pose_goal = {"link_name": "panda_link8", "pose": pose_goal}
+        panda_arm.set_goal(pose_goal=pose_goal)
+
+        # plan to goal
+        logger.info("Planning trajectory")
+        plan_result = panda_arm.plan()
+
+        # execute the plan
+        if plan_result:
+                logger.info("Executing plan")
+                panda_arm.execute()
+
+Single Pipeline Planning - Custom Constraints
+==================================================
+Arguably the most flexible way to specify a goal which we wish to plan for is via custom constraints, here we demonstrate planning to a configuration that satisfies a set of joint constraints: ::
```

Comment:
I'd leave out the subjective wording.

```suggestion
You can also control the output of motion planning via custom constraints. Here we demonstrate planning to a configuration that satisfies a set of joint constraints: ::
```

If there is a doc link to constraints, would be nice to have here too.